lenskit-eval/src/main/java/org/lenskit/eval/traintest/DataSet.java

```diff
@@ -89,8 +97,11 @@ public DataSet(@Nonnull String name,
         } else {
             attributes = ImmutableMap.copyOf(attrs);
         }
+        this.entityTypes = entityTypes;
```

Comment:
We should copy the list, using `ImmutableList.copyOf(entityTypes)`. Just like we do above for the attributes.